### ISSUE TITLE
Regression: Specified markdown mainpage '...md' has not been defined as input file

### ISSUE THREAD

**Issue by melroy89** [doxygen/doxygen#10110]
**Describe the bug**

**Regression** impact:

```sh
error: Specified markdown mainpage '../doxygen_mainpage.md' has not been defined as input file (warning treated as error, aborting now)
```

However, I did specify this file also as `INPUT=...`. See my config below.

With an older Doxygen version the build went fine. And yes, the `../` is intentional, since I'm within a sub-folder (`build` directory).

**Expected behavior**

No error message and successful Doxygen build.

**Screenshots**
![image](https://github.com/doxygen/doxygen/assets/628926/9fe02c97-315b-4273-86ce-799bb546dee7)

**To Reproduce**

1. Upgrade to the latest Doxygen version (=1.9.7).
2. Use a custom markdown mainpage using `USE_MDFILE_AS_MAINPAGE`
3. Provide the markdown also within the `INPUT`
4. Error!

See config: [Doxyfile.conf.tar.gz](https://github.com/doxygen/doxygen/files/11646636/Doxyfile.conf.tar.gz)

**Version**

Doxygen v1.9.7 (latest atm)
  
**Additional context**

v1.9.3 used to work fine, until this binary was not available anymore, so I needed to upgrade [see commit](https://github.com/vasild/cpp-ipfs-http-client/commit/0226d43f65f4a18acf01ee950a5d7a494023d522).

Aside from the screenshot, you can see the same error appear in GitHub Actions: https://github.com/vasild/cpp-ipfs-http-client/actions/runs/5171107038/jobs/9314441295

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10112 

You can overcome the problem by using in your Doxyfile constructs like: `@CMAKE_SOURCE_DIR@/doxygen_mainpage.md` and use the construct `configure_file` of CMake. This makes your Doxyfile also independent of the place where the build directory is located.

**Comment by melroy89** [comment]
Thank you for the PR!

Yea, something like this: https://github.com/Kitware/CMake/blob/master/Utilities/Doxygen/CMakeLists.txt#L70 and this: https://github.com/Kitware/CMake/blob/master/Utilities/Doxygen/doxyfile.in#L45

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by melroy89** [comment]
Sorry..

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
